### PR TITLE
Handle inaccurate shape in EventDescriptor.

### PR DESCRIPTION
**Pull request by danielballan**
This is due to this line in ophyd: https://github.com/bluesky/ophyd/blob/5942d63fef5d10eab1769143eab23c302746333e/ophyd/areadetector/detectors.py#L74

### CODE REVIEW

File: bluesky_browser/viewer/image.py [hunk 1]
```diff
@@ -77,10 +81,37 @@ def subfactory(self, name, descriptor_doc):
         image_keys = {}
         for key, data_key in descriptor_doc['data_keys'].items():
             ndim = len(data_key['shape'] or [])
+            # We want to record a shape that will match the arr.shape
+            # of the arrays we will see later. Ophyd has been writing
+            # incorrect info into descriptors. We try to detect and corect
```
Comment (mrakitin) [inline bluesky_browser/viewer/image.py]:
```suggestion
            # incorrect info into descriptors. We try to detect and correct
```

File: bluesky_browser/viewer/image.py [hunk 2]
```diff
@@ -77,10 +81,37 @@ def subfactory(self, name, descriptor_doc):
...
+            # that here.
             if ndim == 2:
                 image_keys[key] = data_key['shape']
             elif ndim == 3:
-                image_keys[key] = data_key['shape'][1:]
+                # ophyd <1.4.0 gives (x, y, z) where z is 0
+                # Maybe the better way to detect this is start['version']['ophyd'].
```
Comment (mrakitin) [inline bluesky_browser/viewer/image.py:92]: Shouldn't it be `versions` as in https://github.com/bluesky/bluesky/blob/5d19bb1f77a9fd7062b248a33ac7f80136af464f/bluesky/run_engine.py#L237-L241?

**Review by mrakitin — COMMENTED** [review]
Oh, that's a great fix! Suggested a couple of edits.

File: bluesky_browser/viewer/image.py (hunk 2 above)
Comment (danielballan) [reply bluesky_browser/viewer/image.py:92]: Yes. Fixed.

**Comment by danielballan** [comment]
@mrakitin gave his blessing in person to go ahead with this.

File: bluesky_browser/viewer/image.py (hunk 2 above)
Comment (mrakitin) [reply bluesky_browser/viewer/image.py:92]: Ups, I just checked and don't see it. Not a big deal as it's a comment for now, but if used directly, would be nice to have a valid field name.

**Pull request merged**